src/BuildConfig.cc

```diff
@@ -793,9 +812,18 @@ configureBuild(BuildConfig& config, const bool isDebug) {
   std::vector<fs::path> sourceFilePaths = listSourceFilePaths("src");
   std::string srcs;
   for (fs::path& sourceFilePath : sourceFilePaths) {
+    if (const auto& s = sourceFilePath.filename().string();
+        sourceFilePath != mainSource
+        && s.substr(0, s.find_last_of('.')) == "main") {
+      logger::warn(
+          "source file named main was found not directly in src directory"
```

Comment:
This warning could be emitted multiple times, so specifying the exact path would make more sense, e.g.,:

```suggestion
          fmt::format("`{}` is named `main` but is not located directly in the `src/` directory. This file will not be treated as the program's entry point.", sourceFilePath.string())
```

Feel free to modify this if needed.